### PR TITLE
Add support for index annotations

### PR DESCRIPTION
**Pull request by sywesk**
This PR adds support for index annotations as needed by #1545 

Discussed with @a8m

### CODE REVIEW

**Review by a8m — APPROVED** [review]
Looks great! Please fix the minor comment and I’ll merge this.

File: schema/index/annotation.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
+// Copyright 2019-present Facebook Inc. All rights reserved.
```
Comment (a8m) [inline schema/index/annotation.go:1]: @sywesk, as mentioned in previous review. Please remove this file from this PR, and we'll add this logic in next one inside the entsql package.

**Comment by a8m** [comment]
CI is failing - https://github.com/ent/ent/pull/1562/checks?check_run_id=2600920374

Please run `go generate ./...` from `entc/load` and commit the changes (bindata.go).

File: schema/index/annotation.go (hunk 1 above)
Comment (sywesk) [reply schema/index/annotation.go]: For those reading this: 
- schema/*/annotation is the place for generic annotations
- entsql is for the database specific annotations (and this is one)

**Comment by a8m** [comment]
Thanks for your contribution @sywesk. It's appreciated 🙏

**Comment by a8m** [comment]
@all-contributors please add @sywesk for code

**Pull request merged**